services/jenkins/jenkins-build.service.js

```diff
@@ -1,107 +1,95 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
-const serverSecrets = require('../../lib/server-secrets')
+const Joi = require('joi')
+const { renderBuildStatusBadge } = require('../build-status')
+const JenkinsBase = require('./jenkins-base')
+const {
+  buildTreeParamQueryString,
+  buildUrl,
+  queryParamSchema,
+} = require('./jenkins-common')
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class JenkinsBuild extends LegacyService {
+// https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/model/BallColor.java#L56
+const colorStatusMap = {
+  red: 'failing',
+  red_anime: 'building',
+  yellow: 'unstable',
+  yellow_anime: 'building',
+  blue: 'passing',
+  blue_anime: 'building',
+  green: 'passing',
+  green_anime: 'building',
+  grey: 'not built',
+  grey_anime: 'building',
+  disabled: 'not built',
+  disabled_anime: 'building',
+  aborted: 'not built',
+  aborted_anime: 'building',
+  notbuilt: 'not built',
+  notbuilt_anime: 'building',
+}
+
+const schema = Joi.object({
+  color: Joi.allow(...Object.keys(colorStatusMap)).required(),
+}).required()
+
+module.exports = class JenkinsBuild extends JenkinsBase {
   static get category() {
     return 'build'
   }
 
+  static get defaultBadgeData() {
+    return {
+      label: 'build',
+    }
+  }
+
   static get route() {
     return {
       base: 'jenkins/s',
-      pattern: ':scheme(http|https)?/:host/:job*',
+      pattern: ':protocol(http|https)/:host/:job+',
+      queryParamSchema,
     }
   }
 
   static get examples() {
     return [
       {
         title: 'Jenkins',
-        pattern: ':scheme/:host/:job',
         namedParams: {
-          scheme: 'https',
+          protocol: 'https',
           host: 'jenkins.qa.ubuntu.com',
```

Comment:
I am also intentionally not including the new `disableStrictSSL` query param here as I do not believe this is something we want to advertise; it's a work around for a handful of exceptional circumstances only.